diff --git a/window_tiler.py b/window_tiler.py
index 9bbbdf4..969a9bf 100755
--- a/window_tiler.py
+++ b/window_tiler.py
@@ -37,7 +37,7 @@ def get_active_window_id() -> str:
     """
     Get window ID of currently active window
     """
-    return subprocess.check_output(['xdotool', 'getactivewindow']).decode().strip()
+    return subprocess.check_output(['@xdotool@', 'getactivewindow']).decode().strip()
 
 
 def reset_window():
@@ -45,7 +45,7 @@ def reset_window():
     Reset window maximizations so move-resize actually works
     """
     subprocess.call(
-        ['wmctrl', '-i', '-r', get_active_window_id(), '-b', 'remove,maximized_vert,maximized_horz'])
+        ['@wmctrl@', '-i', '-r', get_active_window_id(), '-b', 'remove,maximized_vert,maximized_horz'])
 
 
 def window_move_resize(x: int, y: int, width: int, height: int):
@@ -53,7 +53,7 @@ def window_move_resize(x: int, y: int, width: int, height: int):
     Move and resize window
     """
     reset_window()
-    cmd = ['wmctrl', '-i', '-r', get_active_window_id(), '-e', f'1,{x},{y},{width},{height}']
+    cmd = ['@wmctrl@', '-i', '-r', get_active_window_id(), '-e', f'1,{x},{y},{width},{height}']
     logger.debug(f"Running cmd: '{' '.join(cmd)}'")
     return subprocess.check_output(cmd)
 
@@ -73,7 +73,7 @@ def get_monitor(x) -> Monitor:
 
 
 def get_window_dimensions() -> WindowDimension:
-    win_info = subprocess.check_output(['xwininfo', '-id', get_active_window_id()]).decode()
+    win_info = subprocess.check_output(['@xwininfo@', '-id', get_active_window_id()]).decode()
     logger.debug(win_info)
     abs_left_x = int(re.findall('Absolute upper-left X:  (\d*)', win_info)[0])
     abs_left_y = int(re.findall('Absolute upper-left Y:  (\d*)', win_info)[0])
