diff --git a/editorconfig.el b/editorconfig.el
index eb56df6..aae5f02 100644
--- a/editorconfig.el
+++ b/editorconfig.el
@@ -277,6 +277,11 @@ properties."
   :type '(repeat string)
   :group 'editorconfig)
 
+(defcustom editorconfig-use-ws-butler nil
+  "Use command `ws-butler-mode' for trimming trailing whitespace."
+  :type 'boolean
+  :group 'editorconfig)
+
 (defvar editorconfig-properties-hash nil
   "Hash object of EditorConfig properties for current buffer.
 Set by `editorconfig-apply' and nil if that is not invoked in
@@ -302,6 +307,11 @@ number - `lisp-indent-offset' is not set only if indent_size is
   (and (stringp string)
        (string-match-p "\\`[0-9]+\\'" string)))
 
+(defun editorconfig-use-ws-butler-p ()
+  "Return non-nil if `editorconfig-use-ws-butler' is enabled and
+`ws-butler' is available."
+  (and editorconfig-use-ws-butler (fboundp 'ws-butler-mode)))
+
 (defun editorconfig-set-indentation/python-mode (size)
   "Set `python-mode' indent size to SIZE."
   (set (make-local-variable (if (or (> emacs-major-version 24)
@@ -424,18 +434,22 @@ TRIM-TRAILING-WS."
              (not buffer-read-only))
     ;; when true we push delete-trailing-whitespace (emacs > 21)
     ;; to write-file-functions
-    (add-to-list
-     'write-file-functions
-     'delete-trailing-whitespace))
+    (if (editorconfig-use-ws-butler-p)
+        (ws-butler-mode 1)
+      (add-to-list
+       'write-file-functions
+       'delete-trailing-whitespace)))
   (when (or (equal trim-trailing-ws "false")
             buffer-read-only)
     ;; when false we remove every delete-trailing-whitespace
     ;; from write-file-functions
-    (setq
-     write-file-functions
-     (delete
-      'delete-trailing-whitespace
-      write-file-functions))))
+    (if (editorconfig-use-ws-butler-p)
+        (ws-butler-mode 0)
+      (setq
+       write-file-functions
+       (delete
+        'delete-trailing-whitespace
+        write-file-functions)))))
 
 (defun editorconfig-set-line-length (length)
   "Set the max line length (`fill-column') to LENGTH."
